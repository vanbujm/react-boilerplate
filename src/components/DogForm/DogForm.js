import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import { compose, withHandlers, withState } from 'recompose';
import { camelCase, capitalize } from 'lodash/string';
import s from './DogForm.css';

const formFields = ['name', 'breed'];

export class DogFormComponent extends React.Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    ...Object.assign(
      {},
      ...formFields.map(field => {
        const obj = {};
        obj[field] = PropTypes.string.isRequired;
        return obj;
      }),
    ),
  };

  render() {
    const { onSubmit, onChange } = this.props;
    const textInputs = formFields.map(field => (
      <label htmlFor={field} key={field}>
        {capitalize(field)}:
        <input
          id={field}
          type="text"
          value={this.props[field]}
          onChange={onChange}
        />
      </label>
    ));
    return (
      <form onSubmit={onSubmit} className={s.root}>
        {textInputs}
        <input type="submit" value="Submit" />
      </form>
    );
  }
}

const createUpdateFunction = functionName =>
  camelCase(`update ${functionName}`);

const handlerFunctions = {
  onSubmit: () => e => {
    e.preventDefault();
  },

  onChange: props => e => {
    const updateFunction = createUpdateFunction(e.target.id);
    props[updateFunction](e.target.value);
  },
};

export default compose(
  ...formFields.map(field => withState(field, createUpdateFunction(field), '')),
  withHandlers(handlerFunctions),
  withStyles(s),
)(DogFormComponent);
