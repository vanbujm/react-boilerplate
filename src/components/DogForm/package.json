{
  "name": "DogForm",
  "version": "0.0.0",
  "private": true,
  "main": "./DogForm"
}
