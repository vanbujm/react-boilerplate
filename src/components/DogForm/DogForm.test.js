/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { DogFormComponent } from './DogForm';

describe('AddDog', () => {
  test('renders correctly', () => {
    const onSubmit = () => {};
    const onChange = () => {};
    const updateName = () => {};
    const updateBreed = () => {};
    const name = '';
    const breed = '';

    const wrapper = renderer
      .create(
        <DogFormComponent
          onSubmit={onSubmit}
          onChange={onChange}
          name={name}
          breed={breed}
          updateName={updateName}
          updateBreed={updateBreed}
        />,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
