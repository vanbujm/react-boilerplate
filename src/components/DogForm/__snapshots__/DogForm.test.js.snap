// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDog renders correctly 1`] = `
<form
  className="root"
  onSubmit={[Function]}
>
  <label
    htmlFor="name"
  >
    Name
    :
    <input
      id="name"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <label
    htmlFor="breed"
  >
    Breed
    :
    <input
      id="breed"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
